### PR TITLE
1040 Adjust health checks for ECS

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#1040

### Dependencies

- Upstream: https://github.com/metriport/fhir-server/pull/65
- Downstream: none

### Description

Adjust health checks for ECS - context on each change.

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: infra/lib/fhir-server-stack.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export function settings(): Settings {
     return {
       ...defaults,
       cpu: 1 * vCPU,
-      memoryLimitMiB: 1024,
+      memoryLimitMiB: 2048,
```
Comment (leite08) [inline infra/lib/fhir-server-stack.ts:68]: 1 vCPU requires at least 2GB of RAM - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_size

File: infra/lib/fhir-server-stack.ts [hunk 2]
```diff
@@ -74,8 +74,8 @@ export function settings(): Settings {
   }
   return {
     ...defaults,
-    cpu: 0.5 * vCPU,
-    memoryLimitMiB: 1024,
+    cpu: 1 * vCPU,
+    memoryLimitMiB: 2048,
```
Comment (leite08) [inline infra/lib/fhir-server-stack.ts:78]: 0.5 vCPU was not being enough to start the service - health checks were failing, even adjusting those thresholds - [context](https://metriport.slack.com/archives/C04DMKE9DME/p1722359008629989?thread_ts=1722348020.483229&cid=C04DMKE9DME)

File: infra/lib/fhir-server-stack.ts [hunk 3]
```diff
@@ -291,7 +291,9 @@ export class FHIRServerStack extends Stack {
     // See for details: https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-health-checks.html
     fargateService.targetGroup.configureHealthCheck({
       healthyThresholdCount: 2,
-      interval: Duration.seconds(30),
+      unhealthyThresholdCount: 4,
+      interval: Duration.seconds(20),
+      timeout: Duration.seconds(15),
```
Comment (leite08) [inline infra/lib/fhir-server-stack.ts:296]: https://metriport.slack.com/archives/C04DMKE9DME/p1722372758459789?thread_ts=1722348020.483229&cid=C04DMKE9DME

File: infra/lib/fhir-server-stack.ts [hunk 4]
```diff
@@ -53,8 +53,8 @@ export function settings(): Settings {
   if (isProd(config)) {
     return {
       ...defaults,
-      cpu: 2 * vCPU,
-      memoryLimitMiB: 4096,
+      cpu: 4 * vCPU,
+      memoryLimitMiB: 8192,
```
Comment (leite08) [inline infra/lib/fhir-server-stack.ts:57]: Actually, considering the [original comment](https://github.com/metriport/fhir-server/pull/65/files#r1696169664) and the [weekly load of max CPU](https://us-west-1.console.aws.amazon.com/cloudwatch/home?region=us-west-1#metricsV2?graph=~(metrics~(~(~'AWS*2fECS~'CPUUtilization~'ServiceName~'FHIRServerStack-FHIRServerFargateService46BCF377-gXh9molzfoo4~'ClusterName~'FHIRServerStack-FHIRServerClusterC4EE8AC0-00Vo19v9bvc5~(region~'us-west-1))~(~'.~'MemoryUtilization~'.~'.~'.~'.~(region~'us-west-1~yAxis~'right)))~period~300~region~'us-west-1~stacked~false~title~'CPU*20utilization~view~'timeSeries~stat~'Maximum~start~'-PT168H~end~'P0D)), taking a closer look at some of the spikes, we see they run for ~100% for minutes, so its clear the CPU is also a bottleneck - see below:

Jul 27:

image

Jul 26

image

Jul 29 (single spike)

image

**Review by jonahkaye — APPROVED** [review]
Looks good

**Pull request merged**